DigitalLearningSolutions.Data/Services/LearningHubResourceService.cs

```diff
@@ -0,0 +1,103 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.ApiClients;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Models.External.LearningHubApiClient;
+    using DigitalLearningSolutions.Data.Models.LearningResources;
+    using Microsoft.Extensions.Logging;
+
+    public interface ILearningHubResourceService
+    {
+        Task<LearningResourceReferenceWithResourceDetails?> GetResourceByReferenceId(int resourceReferenceId);
+
+        Task<BulkLearningResourceReferences> GetBulkResourcesByReferenceIds(
+            IList<int> resourceReferenceIds
+        );
+    }
+
+    public class LearningHubResourceService : ILearningHubResourceService
+    {
+        private readonly ILearningHubApiClient learningHubApiClient;
+        private readonly ILearningResourceReferenceDataService learningResourceReferenceDataService;
+        private readonly ILogger<LearningHubResourceService> logger;
+
+        public LearningHubResourceService(
+            ILearningHubApiClient learningHubApiClient,
+            ILearningResourceReferenceDataService learningResourceReferenceDataService,
+            ILogger<LearningHubResourceService> logger
+        )
+        {
+            this.learningHubApiClient = learningHubApiClient;
+            this.learningResourceReferenceDataService = learningResourceReferenceDataService;
+            this.logger = logger;
+        }
+
+        public async Task<LearningResourceReferenceWithResourceDetails?> GetResourceByReferenceId(
+            int resourceReferenceId
+        )
+        {
+            try
+            {
+                // TODO HEEDLS-747: tackle different response codes rather than catching all exceptions

```

Comment:
Just what I was about to say. I totally agree, but think it's worth doing a bit more of the work in this ticket. It's a bit scary to be relying on response-parsing exceptions to tell us whether our request has been successful. Instead, can we explicitly throw an exception (some type name saying "we got a bad response" - maybe `LearningHubResponseException`) if we don't get a 200 back? Then we know what we're catching here.